.golangci.yml

```diff
@@ -46,23 +46,50 @@ output:
 
 linters:
   enable:
+    - deadcode
     - megacheck
     - govet
     - golint
     - interfacer
     - unconvert
     - dupl
     - goconst
+    - gocritic
     - gocyclo
     - goimports
+    - interfacer
     - maligned
     - megacheck
+    - scopelint
+    - staticcheck
+    - structcheck
     - misspell
     - unparam
     - nakedret
     - bodyclose
 
+linters-settings:
+  gocritic:
+    enabled-tags:
+      - performance
+      - opinionated
+      - diagnostic
+    disabled-checks:
```

Comment:
Did these checks produce too many diffs or did they produce diffs that you didn't agree with?

Reply:
I don't like paramTypeCombine especially in this case:
```
            ^
pkg/gameservers/controller.go:91:1: paramTypeCombine: func(
        wh *webhooks.WebHook,
        health healthcheck.Handler,
        minPort, maxPort int32,
        sidecarImage string,
        alwaysPullSidecarImage bool,
        sidecarCPURequest resource.Quantity,
        sidecarCPULimit resource.Quantity,
        sdkServiceAccount string,
        kubeClient kubernetes.Interface,
        kubeInformerFactory informers.SharedInformerFactory,
        extClient extclientset.Interface,
        agonesClient versioned.Interface,
        agonesInformerFactory externalversions.SharedInformerFactory) *Controller could be replaced with func(wh *webhooks.WebHook,
        health healthcheck.Handler,
        minPort, maxPort int32,
        sidecarImage string,
        alwaysPullSidecarImage bool,
        sidecarCPURequest,
        sidecarCPULimit resource.Quantity,

        sdkServiceAccount string,
        kubeClient kubernetes.Interface,
        kubeInformerFactory informers.SharedInformerFactory,
        extClient extclientset.Interface,
        agonesClient versioned.Interface,
        agonesInformerFactory externalversions.SharedInformerFactory) *Controller (gocritic)
```
Seems it just combined `sidecarCPURequest, sidecarCPULimit resource.Quantity,`, but I can add this check also if it is needed.

Reply:
`sloppyReassign` example:
```
pkg/gameservers/controller.go:389:5: sloppyReassign: re-assignment to `err` can be replaced with `err := c.syncGameServerShutdownState(gs)` (gocritic)
        if err = c.syncGameServerShutdownState(gs); err != nil {
```

Reply:
For the paramTypeCombine check, the common suggestion I've heard is that when parameter lists get that long you should create a struct and pass the struct instead of a list of individual parameters. The reason being that it adds better type checking, and when the parameter list gets long it becomes easy to pass things in the wrong order, especially if parameters have the same type. This check seems to make that worse, so I agree that we shouldn't enable it. 

The change in sloppyReassign looks correct to me.